extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -43,21 +44,31 @@ export class AmpExperiment extends AMP.BaseElement {
 
     const config = this.getConfig_();
     const results = Object.create(null);
+    const override = this.getVariantOverride_();
     const variants = Object.keys(config).map(experimentName => {
+      if (override[experimentName]) {
+        results[experimentName] = override[experimentName];
```

Comment:
I don't think the risk is eliminated by limiting the characters. Do we really need to allow overrides to be a different value than one of the predefined variants? If so, I think the risk is much greater than the benefit. There are many kinds of attacks this enables. Here is another example:

a publisher is experimenting with an image for a sensitive news article:

headline-experiment: {
  variant: {
    "sadImage": 50
    "evenSadderImage": 50
  }
}
<amp-img src="img/VARIANT(headline-experiment).jpg"></amp-img>

now an attacker can find an inappropriate image that already exists on publisher's image directory and share a link like

http://publisher.com/sensitive/topic/amp?amp-x-headline-experiment=totallyInAppropriateImage
in social media. Now, the top-level domain is still publisher.com so readers will be trusting the content is indeed by `publisher.com` and that can go viral hurting the publisher.

I feel very strongly that we should not allow override values that are not already defined by the page author as variants.